### PR TITLE
Add additional parameters to hisat2 process

### PR DESCRIPTION
**Pull request by jvrakor**
@mstajdohar @jkokosar @dblenkus

### CODE REVIEW

File: docs/CHANGELOG.rst [hunk 1]
```diff
@@ -12,6 +12,16 @@ Unreleased
 
 Added
 -----
+* Add `Non-canonical splice sites penalty`, `Disallow soft clipping`
+  and `Report alignments tailored specifically for Cufflinks`
+  parameters to hisat2 process
+
+================
```
Comment (dblenkus) [inline docs/CHANGELOG.rst]: Why is this added?

File: resolwe_bio/processes/alignment/hisat2.yml [hunk 2]
```diff
@@ -58,13 +74,32 @@
...
 
       {% else %}
 
-        hisat2 -p {{threads}} --no-unal -x ${INDEX} -1 ${FW_READS} -2 ${RW_READS} -S alignment_unsorted.sam 2> stats.txt
+        hisat2 \
+          -p {{threads}} \
+          --no-unal \
+          -x ${INDEX} \
+          -1 ${FW_READS} \
+          -2 ${RW_READS} \
```
Comment (dblenkus) [inline resolwe_bio/processes/alignment/hisat2.yml]: This is the only line that is different in two commands. Please combine them to one and use if for this line.

File: resolwe_bio/processes/alignment/hisat2.yml [hunk 3]
```diff
@@ -58,13 +74,32 @@
...
+          {% if cufflinks %}--dta-cufflinks{% endif %} \
+          2> stats.txt
         re-checkrc "HISAT2 alignment failed."
         re-progress 0.6
 
       {% else %}
 
-        hisat2 -p {{threads}} --no-unal -x ${INDEX} -1 ${FW_READS} -2 ${RW_READS} -S alignment_unsorted.sam 2> stats.txt
+        hisat2 \
+          -p {{threads}} \
```
Comment (dblenkus) [inline resolwe_bio/processes/alignment/hisat2.yml]: Threads setting doesn't work any more. Discuss with @jkokosar how to change it.

File: resolwe_bio/processes/alignment/hisat2.yml [hunk 4]
```diff
@@ -8,7 +8,7 @@
   requirements:
     expression-engine: jinja
   data_name: "Alignment ({{ reads|sample_name|default('?') }})"
-  version: 1.0.7
+  version: 1.0.8
```
Comment (dblenkus) [inline resolwe_bio/processes/alignment/hisat2.yml]: We should discuss which number in version to increase for which changes.

This change introduces new features, but is backward compatible. So I think middle number should be increased.

cc @tjanez, @mstajdohar, @jkokosar

File: docs/CHANGELOG.rst (hunk 1 above)
Comment (jkokosar) [inline docs/CHANGELOG.rst]: Should be under "Changed". Also, rebase will make an invalid merge of changelog. Keep an eye on that.

File: resolwe_bio/processes/alignment/hisat2.yml (hunk 3 above)
Comment (jkokosar) [inline resolwe_bio/processes/alignment/hisat2.yml]: @jvrakor Example of how things should be changed:
https://github.com/genialis/resolwe-bio/blob/master/resolwe_bio/processes/alignment/bowtie.yml#L295
https://github.com/genialis/resolwe-bio/blob/master/resolwe_bio/processes/alignment/bowtie.yml#L559
